### ISSUE TITLE
JTR tests may need to call `MerkleDb.resetDefaultInstancePath()`

### ISSUE THREAD

**Issue by anthony-swirldslabs** [hashgraph/hedera-services#17001]
**Background**
`AddressBook-restart-upgrade-add-node-1k-4m` fails with `IllegalStateException: Table already exists: RosterService.ROSTERS`. This indicates that it tries to initialize a state while a previous state has already initialized a virtual map for rosters.

In unit test, we usually call `MerkleDb.resetDefaultInstancePath()` to work around this.

**Acceptance criteria**
Ensure the second ROSTERS map is created at a different path, probably using a call to `MerkleDb.resetDefaultInstancePath()`.

**Comment by edward-swirldslabs** [comment]
For the JRS address book tests, the nodes are being run in different VMs.  This would not be a clash with a sibling node on the same VM.  Given the address book is doing a restart, perhaps the node is binding to the temp file created when it first ran after it restarts?

**Comment by anthony-swirldslabs** [comment]
If the node restarts, it should be able to read the old saved state and initialize the ROSTERS map **once**. This exception should only pop up if this map has already been initialized in the running JVM previously, and the code tries to create a new VirtualMap instance with name ROSTERS w/o a call to `resetDefaultInstancePath`. This can happen, for example, if the code loads another saved state from disk or obtains a new state via reconnect.

**Comment by edward-swirldslabs** [comment]
Yep, it was node 4, the one added after upgrade, performing a reconnect.

**Comment by anthony-swirldslabs** [comment]
That's interesting. For the reconnect process, I've added the following:

https://github.com/hashgraph/hedera-services/blob/4f9e245f30737f50ed5610c11ba14dcfe43e8109/platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/reconnect/ReconnectLearner.java#L162

to unregister the states before passing the state to the system for adopting it as an actual state. Before that, the states are registered upon receiving the state at:

https://github.com/hashgraph/hedera-services/blob/4f9e245f30737f50ed5610c11ba14dcfe43e8109/platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/reconnect/ReconnectLearner.java#L216

Existing JRS Panel tests that verify the reconnect behavior seemed to be happy with this approach.

I'm wondering as to what is different in this particular JTR test that causes it to fail...
